dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/PublishDatasetCommand.java

```diff
@@ -63,6 +65,14 @@ public PublishDatasetResult execute(CommandContext ctxt) {
 
         Dataset theDataset = getDataset();
 
+        String protocol = getDataset().getProtocol();
+        GlobalIdServiceBean idServiceBean = GlobalIdServiceBean.getBean(protocol, ctxt);
+        boolean reservingPidsSupported = !idServiceBean.registerWhenPublished();
```

Comment:
I feel that this line needs a comment, as it's a bit unclear why this works that way.